src/main/scala/akka/io/AsyncDnsResolver.scala

```diff
@@ -9,32 +11,35 @@ import ru.smslv.akka.dns.raw.{AAAARecord, ARecord, Answer, CNAMERecord, DnsClien
 
 import scala.annotation.tailrec
 import scala.collection.JavaConversions._
-import scala.collection.{breakOut, immutable, mutable}
+import scala.collection.{breakOut, immutable}
+import scala.concurrent.duration.{Deadline, FiniteDuration}
+import scala.io.Source
 import scala.util.Random
-
-private case class CurrentRequest(client: ActorRef,
-                                  name: String,
-                                  ipv4: Option[immutable.Seq[Inet4Address]],
-                                  ipv6: Option[immutable.Seq[Inet6Address]],
-                                  srv: Option[immutable.Seq[SRVRecord]],
-                                  ttl: Long)
+import scala.util.control.NonFatal
 
 class AsyncDnsResolver(cache: SimpleDnsCache, config: Config) extends Actor with ActorLogging {
 
   import AsyncDnsResolver._
 
-  private val nameServers: immutable.Seq[InetSocketAddress] = config.getStringList("nameservers").map(parseNameserverAddress)(breakOut)
+  private val systemNameServers =
```

Comment:
That argument I don't buy, however, it won't compile since parseSystemNameServers returns an option in-and-of itself. :) (which is an argument I'd buy ) :)